nilearn/glm/first_level/hemodynamic_models.py

```diff
@@ -565,9 +565,8 @@ def compute_regressor(exp_condition, hrf_model, frame_times, con_id='cond',
         fir_delays = [int(x) for x in fir_delays]
     oversampling = int(oversampling)
 
-    # this is the average tr in this session, not necessarily the true tr
-    tr = float(frame_times.max()) / (np.size(frame_times) - 1)
-
+    # this is the minimal tr in this session, not necessarily the true tr
+    tr = float(np.min(frame_times[1:] - frame_times[:-1]))
```

Comment:
```suggestion
    tr = np.min(np.diff(frame_times))
```

Don't use this if you don't like it, but it's a bit simpler.